src/java.base/share/classes/jdk/internal/access/JavaLangAccess.java

```diff
@@ -305,11 +307,12 @@ public interface JavaLangAccess {
     int countNonZeroAscii(String s);
 
     /**
-     * Constructs a new {@code String} by decoding the specified subarray of
-     * bytes using the specified {@linkplain java.nio.charset.Charset charset}.
-     *
-     * The caller of this method shall relinquish and transfer the ownership of
-     * the byte array to the callee since the later will not make a copy.
+     * Constructs a new {@code String} by decoding the specified byte array
+     * using the specified {@linkplain java.nio.charset.Charset charset}.
+     * <p>
+     * <b>WARNING: The caller of this method shall relinquish and transfer the
+     * ownership of the byte array to the callee</b>, since the later will not
+     * make a copy.
```

Comment:
I think we should add a note on why no copy is made - there may be multiple reads to the input array for decoding, so the input array must not be accessible by untrusted code, which can racily modify it. A side effect is that this array is simply reused if it's eligible to be the string content array, which is unfortunately abused in a few places in the JDK.

Reply:
@liach, thank you so much for the review and the feedback. We internally discussed this and decided leave the change as is. Resolving this conversation. Please let me know if you think otherwise.